src/server/action_process_queue.ts

```diff
@@ -0,0 +1,52 @@
+import * as spawn from "child_process"
+import * as express from "express"
+import * as Pq from "p-queue"
+import * as winston from "winston"
+
+export class AsyncProcessJob {
+    queue: Pq
+
+    constructor() {
+        this.queue = new Pq({concurrency: 1})
+    }
+
+    async asyncProcess(req: express.Request, res: express.Response) {
```

Comment:
Can we generalize this with something like:

- Making this an abstract class "ExternalProcessJob" which has a `run` function that takes JSON input and returns JSON
- Have a new "ExecuteJob" extend "ExternalProcessJob" and add a new method that translates the action specific stuff into the JSON for XPC
- Leave anything to do with Express in `server.js` since its not really relevant to the execution process or the forking process